File: dotnet/Allmark/Render/ConsoleHighlightRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ConsoleHighlightRenderer
{
    public static Renderer Create(string style, string reset)
    {
        return new Renderer
        {
            Name = "highlight",
            Render = (node, state, first, last, decode) => Render(node, state, style, reset),
        };
    }

    public static Renderer Create()
    {
        return new Renderer
        {
            Name = "highlight",
            Render = (node, state, first, last, decode) => Render(node, state, RenderToConsole.AnsiYellowBg + RenderToConsole.AnsiBlack, RenderToConsole.AnsiReset),
        };
    }

    public static void Render(MarkdownNode node, RendererState state, string style, string reset)
    {
        state.Output.Append(style);
        RenderChildren.Execute(node, state);
        state.Output.Append(reset);
    }
}

File: swift/Sources/allmark/RenderConsole/ConsoleHighlightRenderer.swift
import Foundation

@MainActor
let consoleHighlightRenderer = Renderer(
	name: "highlight",
	render: renderConsoleHighlight
)

@MainActor
func renderConsoleHighlight(_ node: MarkdownNode, _ state: inout RendererState, _ first: Bool?, _ last: Bool?, _ decode: Bool?) {
	state.output += "\u{001B}[43m\u{001B}[30m"
	renderChildren(node: node, state: &state)
	state.output += "\u{001B}[0m"
}
